### PR TITLE
Contracts: DerivedPriceFeed updates

### PR DESCRIPTION
**Pull request by Nic0S**
- Extend AggregaotV3Interface since V2 funcs are not needed

 - Make decimals, base, quote immutable

 - In getDerivedPrice, read from contract instead of func args

 - Move into dev/v0.8

 - Small style updates

### CODE REVIEW

File: contracts/src/v0.8/dev/DerivedPriceFeed.sol [hunk 1]
```diff
@@ -18,35 +18,36 @@ import "../interfaces/AggregatorV2V3Interface.sol";
...
-  uint256 public override latestRound;
-
-  mapping(uint256 => int256) public override getAnswer;
-  mapping(uint256 => uint256) public override getTimestamp;
-  mapping(uint256 => uint256) private getStartedAt;
-
-  AggregatorV3Interface public base;
-  AggregatorV3Interface public quote;
+  AggregatorV3Interface public immutable BASE;
+  AggregatorV3Interface public immutable QUOTE;
```
Comment (makramkd) [inline contracts/src/v0.8/dev/DerivedPriceFeed.sol:35]: Is there ever a situation where we'd want to set these to something else? Or are these truly a "set it and forget it" thing?

File: contracts/test/v0.8/dev/DerivedPriceFeed.test.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      let quote = await mockAggFactory.deploy(8, 5e8) // 5
+
+      let derived = await derivedFeedFactory.deploy(
+        base.address,
+        quote.address,
+        8,
+      )
+
+      await expect((await derived.latestRoundData()).answer).to.equal(
+        BigNumber.from('200000000' /* 2 */),
```
Comment (makramkd) [inline contracts/test/v0.8/dev/DerivedPriceFeed.test.ts:59]: Super small nit: can we use scientific notation in the string instead? `2e8`. Easier to read IMO. Similar comments for the rest of the tests.

**Review by makramkd — DISMISSED** [review]

File: contracts/src/v0.8/dev/DerivedPriceFeed.sol (hunk 1 above)
Comment (Nic0S) [reply contracts/src/v0.8/dev/DerivedPriceFeed.sol:35]: You mean like change them after deployment? I don't think so, if we wanted to make them changeable would probably point to a proxy here

Comment (makramkd) [reply contracts/src/v0.8/dev/DerivedPriceFeed.sol:35]: Yeah, I meant after deployment. Sounds good 👍

File: contracts/test/v0.8/dev/DerivedPriceFeed.test.ts (hunk 2 above)
Comment (Nic0S) [reply contracts/test/v0.8/dev/DerivedPriceFeed.test.ts]: BigNumber.from doesn't support that, but I got rid of it where I could and replaced with scientific notation

**Review by cmalec — APPROVED** [review]

**Pull request merged**